src/main/scala/fi/oph/koski/validation/KoskiValidator.scala

```diff
@@ -379,6 +380,27 @@ class KoskiValidator(tutkintoRepository: TutkintoRepository, val koodistoPalvelu
     }
   }
 
+  private def validateOsaAikainenErityisopetus(oo: KoskeenTallennettavaOpiskeluoikeus): HttpStatus = oo match {
+    case t: TukimuodollinenOpiskeluoikeus => {
+      val lt = t.lisätiedotSisältääOsaAikaisenErityisopetuksen
+      val s = t.suoritusSisältääOsaAikaisenErityisopetuksen
+
+      val ehkäVuosiluokan = if (t.isInstanceOf[PerusopetuksenOpiskeluoikeus]) "vuosiluokan " else ""
```

Comment:
Keksiikö joku tähän paremman tavan kuin isInstanceOf:in... ei ainakaan millään huvittaisi tehdä oo-tyyliä tähän pelkkään virheilmoituksen muodostamiseen case-luokkien sekaan.